shpc/main/registry/remote.py

```diff
@@ -138,48 +146,35 @@ def exists(self, name):
         if self._cache and name in self._cache:
             return True
         dirname = self.source
-        if self.subdir:
```

Comment:
Hmm Do we have a test case that triggers this? Do we set a default for self.source? It could be possible that it's either unset or set to something unexpected. Should the behavior differ based on is_cloned?

Reply:
Originally, self.source was first set to the URL, and then turned into the directory of the clone.
With my change, it remains unset until the repository is cloned, and then it's the directory + the sub-directory.

No there isn't a test case for that

Reply:
okay, let's add one more check here then because if for some reason the cache is empty and we haven't cloned yet, it's going to try to os.path.join a None and throw an error. It's up to you what you want to do default to - either it could a warning, or a clone right then, etc,

Reply:
Looking at where `source` is called throughout the codebase, I can indeed see some weaknesses. I'll fix these. Stay tune :)